Automatically handle SHA versions.

samth opened the pull request:
For discussion, needs documentation.

mflatt commented:
Something like this might be a good idea if ci-snapshot evolves to support cross builds — that is, to have builds other than x86_64 Linux. As it is, ci-snapshot only works to install run old builds if you're on x86_64 Linux.

I'm also skeptical about installing snapshots in `$(ADDON_DIR)/raco-cross`, because they'll quietly build up. It may be a good idea to require a `--workspace` argument for a non-regular version.

mflatt committed:
split into -lib and -doc

mflatt committed:
fix README title

mflatt committed:
fix doc typo

mflatt committed:
fix doc typo

mflatt committed:
configure name for each installation

Even though each cross installation is configured to install package
in "installation" scope, there are various ways that a user-scope
directory can get involved, such as packages installed in a normal
installation with the same version (potentially including the one used
to run `raco cross`, and potentially involving the `raco cross`
package itself). Avoid those interactions by giving each cross
installation a distinct name.

Closes mflatt/raco-cross#3

mflatt committed:
use download.racket-lang.org instead of mirror.racket-lang.org

mflatt committed:
set name to `raco` when running `racket -l raco`

Related to #1

mflatt committed:
pin a workspace to a version

Also, record a specified installers URL with a workspace that is
selected by `--workspace`.

Closes #1

mflatt committed:
avoid user-scope use until after installation name is configured

Closes #3

sschwarzer committed:
Change `scribble` to `raco-cross` in URLs

Probably due to a copy/paste mistake the "pull request" and "issue"
links in the README.md were pointing to `racket/scribble` instead of
`racket/raco-cross`.

mflatt committed:
fix version-mismatch reporting

mflatt committed:
update cross-compile steps for new Racket build system

samth committed:
Add test submodule to avoid test failure in pkg-build.
